nbgrader/__init__.py

```diff
@@ -55,6 +55,12 @@ def _jupyter_nbextension_paths():
             dest="validate_assignment",
             require="validate_assignment/main"
         ),
+        dict(
```

Comment:
Hmm, so adding this new extension is going to require a bit more work in terms of updating documentation, demos, etc. in a variety of places than what you've done here. We'll also need to make sure it's compatible with JupyterLab, which may be a bit more work. What I'd suggest for now is to remove the extension code from this PR and made a new PR adding it in. Then we can focus on the autogenerated test functionality alone in this PR, and discuss what else needs to be added for the extension in the other PR.

Reply:
The extension related changes have been removed from this PR. We will make a new PR for that separately.